### PR TITLE
feat(efs): transition to archive for `FileSystem`

### LINKED ISSUE #28720: efs: support `TransitionToArchive` property for `AWS::EFS::FileSystem`
### Describe the feature

Support `TransitionToArchive` property for `AWS::EFS::FileSystem`

More about this property [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-efs-filesystem-lifecyclepolicy.html#cfn-efs-filesystem-lifecyclepolicy-transitiontoarchive)

### Use Case

Currently the property is not supported by `FileSystem` L2 construct

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

v2.121.1

### Environment details (OS name and version, etc.)

Ubuntu 22.04

### PR DESCRIPTION
**Pull request by longtv2222**
This PR adds support for `TransitionToArchive` property for `AWS::EFS::FileSystem`

Also updated `LifecyclePolicy` to add missing values for `TransitionToArchive` and `TransitionToIA`

[Docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-efs-filesystem-lifecyclepolicy.html#cfn-efs-filesystem-lifecyclepolicy-transitiontoarchive)

Closes #28720.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts [hunk 1]
```diff
@@ -211,6 +227,14 @@ export interface FileSystemProps {
...
    */
   readonly outOfInfrequentAccessPolicy?: OutOfInfrequentAccessPolicy;
+
+  /**
+   * The number of days after files were last accessed in primary storage (the Standard storage class) at which to move them to Archive storage.
+   * Metadata operations such as listing the contents of a directory don't count as file access events.
+   *
+   * @default - None. EFS will not transition files to Archive storage class.
+   */
+  readonly transitionToArchive?: LifecyclePolicy;
```
[the remark this answers is not in the document]
Comment (longtv2222) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts]: Agree, renamed the prop to be more align with other policies

File: packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts [hunk 2]
```diff
@@ -42,7 +43,22 @@ export enum LifecyclePolicy {
...
+
+  /**
+   * After 270 days of not being accessed.
+   */
+  AFTER_270_DAYS = 'AFTER_270_DAYS',
+
+  /**
+   * After 365 days of not being accessed.
+   */
+  AFTER_365_DAYS = 'AFTER_365_DAYS',
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: It looks like `TransitionToIA` and `TransitionToArchive` have the same set of enums excelt that `TransitionToArchive` does not allow `AFTER_1_DAY` [source](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-efs-filesystem-lifecyclepolicy.html)

Can you add in a synth-time check that throws an error if `AFTER_1_DAY` is set for `transitionToArchivePolicy`?  Plus a unit test :)

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @longtv2222, this looks great to me except for the one comment below. Thanks for contributing!

File: packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts (hunk 2 above)
Comment (longtv2222) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: It looks to me that `TransitionToIA` and `TransitionToArchive` both allow `AFTER_1_DAY`, the order is a bit off in the docs...
Screenshot 2024-01-16 143108

Comment (longtv2222) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: It looks like they share the same set of enums to me. The order is a bit off for `AFTER_1_DAY`...
Screenshot 2024-01-16 144600

Comment (longtv2222) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: I'm also able to create EFS with `transitionToArchive` set to 1 day as well!

image

Comment (kaizencc) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: 🤦🤦🤦

Comment (kaizencc) [reply packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts:61]: sorry. the order screwed me up and shame on me for expecting the enum list to be in the same chronological order :)

**Review by kaizencc — APPROVED** [review]

**Pull request merged**